addons/base_import/models/base_import.py

```diff
@@ -982,6 +1217,121 @@ def do(self, fields, columns, options, dryrun=False):
+            new_record = []
+            for fields, indexes in mapping_indexes.items():
+                split_fields = fields.split('/')
+                target_field = split_fields[-1]
+
+                # get target_field type (on target model)
+                target_model = self.res_model
+                for field in split_fields:
+                    if field != target_field:  # if not on the last hierarchy level, retarget the model
+                        target_model = self.env[target_model][field]._name
+                field_type = self.env[target_model].fields_get().get(target_field, {}).get('type', '')
+
+                # merge data if necessary
+                if field_type == 'char':
+                    new_record.append(' '.join(record[idx] for idx in indexes if record[idx]))
+                elif field_type == 'text':
+                    new_record.append('\n'.join(record[idx] for idx in indexes if record[idx]))
+                else:
+                    new_record.append(record[indexes[0]])
+
+            merged_data.append(new_record)
+
+        return import_fields, merged_data
+
+    def _handle_fallback_values(self, import_field, merged_data, fallback_values):
+        """
+        If there are fallback values, this method will replace the input file data value if it does not match the
+        possible values for the given field. This is only valid for boolean and selection fields.
+
+        Note: We can consider that we need to retrieve the selection values for all the fields in fallback_values,
+        as if they are present, it's because there was already a conflict during first import run and user had to
+        select a fallback value for the field.
+
+        :param: (list) import_field: ordered list of field that have been matched to import data
+        :param: (list) merged_data: ordered list of values (list) that need to be imported in the given import_fields
+        :param: (dict) fallback_values: contains all the fields that have been tagged by the user to use a specific
+        fallback value in case the value to import does not match values accepted by the field (selection or boolean)
+            E.g.: {
+                'fieldName': {
+                    'fallback_value': fallback_value,
+                    'field_model': field_model,
+                    'field_type': field_type
+                },
+                'state': {
+                    'fallback_value': 'draft',
+                    'field_model': field_model,
+                    'field_type': 'selection'
+                },
+                'active': {
+                    'fallback_value': 'true',
+                    'field_model': field_model,
+                    'field_type': 'boolean'
+                }
+            }
+        """
+        # get all the selection values
+        for field_string in fallback_values:
+            if fallback_values[field_string]['field_type'] == "boolean":
```

Comment:
Well this is only relevant for selection type fields so I would change the condition to `!= 'selection'`

Reply:
#fatigue